### ISSUE TITLE
utils.utils issue on Colab

### ISSUE THREAD

**Issue by chasmonchichi** [eloimoliner/bwe_historical_recordings#3]
I'm anxious to try out the BWE feature after getting a version of the noise-reduction module working (with some difficulty due to the dependencies) and being very impressed by the results. I wasn't clear whether your audio sample page included BWE processing, or if they were only NR.

I didn't have any luck with cloning the repo, so I tried the Colab. it errored on hydra, which was a quick fix, but then can't load util.utils. I added utils with pip, but still no go.

I was two weeks into training my own 78 rpm noise-reduction model with Demucs hybrid when I came across your work. I've tested your NR-only version primarily on jazz 78's, and there are some ringing artifacts in pianos, but otherwise it's a very, very good model. Would like to see it released in a common format so that it could be used with UVR5 or one of the other GUI's.

**Comment by eloimoliner** [comment]
Hi,

Thank you for informing me that the notebook does not work anymore. I will fix it soon once I have some time. 

utils is not a python package that you can pip install, it is a file in the repo with some functions. There may be something wrong with the imports, I should check.

I would be happy to make the denoising model more accessible, but I'm unfamiliar with the GUIs you are referring to. Could you please provide some more info about that?

**Comment by chasmonchichi** [comment]
There are several GUI front ends that work with AI models built around PyTorch/PyAudio. I think the best, and most actively maintained, is UVR5, which started as “Ultimate Vocal Remover” but now supports a variety of model architectures, including some that remove reverb, separate wind instruments, etc. The developer runs an active Discord server: https://discord.gg/zrA3zeUw

I would like to get your BWE version running locally, but I haven’t been able to import the environment. Anaconda navigator says the .yml is malformed, and trying it from the conda command line errors out on the _openmp_mutex dependency (I’m on Windows). Looks like openmp may no longer be available, even via conda forge.

Please update if you fix the Colab - I’d love to hear what EWB can do. I’ve been restoring 78/cassette/reel/vinyl for many years. The model I was training when I discovered yours was based on a dataset that pulled the last 2 seconds from hundreds of 78s in the George Blood archive, culled out any that contained music, then stitched the remaining snips together in random order to create noise tracks to add to clean sources. My “clean” tracks were recordings of 78’s transferred by the late John RT Davies, a renowned jazz 78 archivist, hoping that the bit of noise he left present would keep the model from being too aggressive (ala CEDAR).

**Comment by eloimoliner** [comment]
Thanks for the suggestion. I'll definitely ask in the discord server to integrate the denoiser.

It looks like your training approach is very similar from what I did. I also extracted the noise data from the same collection. However, in my case, I used modern recordings as the clean data. 

I'll let you know when I fix the colab but, unfortunately, I'm quite busy these days so not sure when I'll have time to. I believe it should be just a package version problem.

Regarding the local installation, I didn't test it on windows, only in Linux. But I should try to reinstall it again to see if I can replicate your problems.

**Comment by chasmonchichi** [comment]
I believe I’m close to having the BWE running. I resolved the package issues and am now working on building a launch string for iteration.py directly, as your bash script didn’t yet include a chooser for the checkpoint types (piano/strings/orchestra).

Since your applications focus on classical music, here’s an idea I’ve had for some time that’s really only possible for classical music content. It’s taken from what I believe Andrew Rose of the Pristine Classical label is currently doing as a manual process. The result is similar to your description of BWE, but instead of rebuilding missing harmonics from scratch, it would use fairly extreme equalization to extract what’s already present in the recordings, but attenuated by the 78 medium. 

Basically you would take a modern recording of a certain classical piece, sample the frequency distribution over time, compare that curve to a similar 78 recording of the same piece, and then apply the delta to the 78 using linear-phase EQ. (This might not even require AI, but it’s also possible that AI could do it with excellent precision.)

Andrew Rose is (I think) doing this as a painstaking manual process, and his results are pretty stunning. It would be interesting to see if it could be automated.

**Comment by eloimoliner** [comment]
Hi, 
Sorry for the late response. That is an interesting idea. For this BWE paper I already computed something of this style, I compared the long-term average spectrum of one historical recording with a modern version of it,  but this was just to get a broad idea of how the frequency response would look like. It makes sense that professional engineers are using this approach, but it seems very tedious and there is a limit at which there is something useful  to recover by only inverting the equalization. What I prefer to do is to generate new (coherent) content, but I'm convinced that estimating the frequency response of the recording is an important step.
I'm now working on a slightly different approach where I use a model to jointly estimate the frequency response and regenerate the recording based on it. I'm quite excited with the results, I hope I will have time to publish this soon.

**Comment by chasmonchichi** [comment]
Based on the quality of the work you've released so far, I'm also very excited to see your results. Please let me know if I can assist your project in any way.On 9 Apr 2023, at 1:02 AM, Eloi Moliner Juanpere ***@***.***> wrote:﻿
Hi,
Sorry for the late response. That is an interesting idea. For this BWE paper I already computed something of this style, I compared the long-term average spectrum of one historical recording with a modern version of it,  but this was just to get a broad idea of how the frequency response would look like. It makes sense that professional engineers are using this approach, but it seems very tedious and there is a limit at which there is something useful  to recover by only inverting the equalization. What I prefer to do is to generate new (coherent) content, but I'm convinced that estimating the frequency response of the recording is an important step.
I'm now working on a slightly different approach where I use a model to jointly estimate the frequency response and regenerate the recording based on it. I'm quite excited with the results, I hope I will have time to publish this soon.

**Comment by eloimoliner** [comment]
The bug in the Colab notebook should be fixed now.
